bash-preexec.sh

```diff
@@ -47,7 +47,17 @@ __bp_last_argument_prev_command="$_"
 
 # Command to set our preexec trap. It's invoked once via
 # PROMPT_COMMAND and then removed.
-__bp_trap_install_string="trap '__bp_preexec_invoke_exec \"\$_\"' DEBUG;"
+__bp_trap_install() {
+  local prior_trap=$(trap -p DEBUG | sed "s/[^']*'\(.*\)'[^']*/\1/")
+  if [[ -n "$prior_trap" ]]; then
+    eval '__bp_original_debug_trap() {
+      '"$prior_trap"'
+    }'
+    preexec_functions+=("$__bp_original_debug_trap")
```

Comment:
I think skipping missing functions is more pragmatic and is the actual behavior by preexec/precmd in zsh. I think we should be trying to emulate that experience.